cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,568 @@
+                meta -> true);
+            long startTs = 0;
+            // Gets the latest run info
+            for (ProgramRunId programRunId : runMap.keySet()) {
+              if (startTs < runMap.get(programRunId).getStartTs()
+                && programRunId.getProgram().equals("DataPipelineWorkflow")) {
+                latestProgramRunId = programRunId;
+                latestRunRecordDetail = runMap.get(programRunId);
+                startTs = runMap.get(programRunId).getStartTs();
+              }
+            }
+            if (latestProgramRunId != null) {
+              JsonObject metrics =
+                queryMetrics(
+                  namespaceId,
+                  app.getName(),
+                  latestProgramRunId.getRun(),
+                  workflowName,
+                  applicationPair.getValue().getConfiguration(),
+                  latestRunRecordDetail != null ? latestRunRecordDetail.getStartTs() : 0,
+                  latestRunRecordDetail != null && latestRunRecordDetail.getStopTs() != null
+                    ? latestRunRecordDetail.getStopTs()
+                    : DateTime.now().getMillis());
+              generateLogFileAndRunInfo(
+                latestProgramRunId, latestRunRecordDetail, appFolderPath.toString(), metrics);
+            }
+          }
+        } else if (appId != null) {
+          File appFolderPath = new File(basePath.toString(), appId);
+          if (!appFolderPath.exists()) {
+            appFolderPath.mkdirs();
+          }
+          Pair<ApplicationId, ApplicationDetail> applicationPair =
+            generateApplicationFile(namespaceId, appId, appFolderPath.toString());
+          latestProgramRunId =
+            new ProgramRunId(namespaceId, appId, ProgramType.WORKFLOW, workflowName, runId);
+          latestRunRecordDetail = store.getRun(latestProgramRunId);
+          JsonObject metrics =
+            queryMetrics(
+              namespaceId,
+              appId,
+              runId,
+              workflowName,
+              applicationPair.getValue().getConfiguration(),
+              latestRunRecordDetail != null ? latestRunRecordDetail.getStartTs() : 0,
+              latestRunRecordDetail != null && latestRunRecordDetail.getStopTs() != null
+                ? latestRunRecordDetail.getStopTs()
+                : DateTime.now().getMillis());
+          generateLogFileAndRunInfo(
+            latestProgramRunId, latestRunRecordDetail, appFolderPath.toString(), metrics);
+        }
+      }
+    } catch (Exception e) {
+      LOG.error("Can not generate support bundle: ", e);
+    } finally {
+      try {
+        if (statusJson != null) {
+          statusJson.close();
+        }
+      } catch (IOException e) {
+        LOG.error("Can not close status json file ", e);
+      }
+    }
+  }
+
+  /** Gets oldest folder from the root directory */
+  private File getOldestFolder(File baseDirectory) {
+    File[] supportFiles = baseDirectory.listFiles();
```

Comment:
Use `DirUtils.listFiles()` instead and use stream to simplify the implementation.

```
return DirUtils.listFiles(baseDirectory).stream()
  .reduce((f1, f2) -> f1.lastModified() < f2.lastModified() ? f1 : f2)
  .orElse(null);
```